backend/users/views/user.py

```diff
@@ -130,57 +102,59 @@ def get_queryset(self):
     def create(self, request, *args, **kwargs):
         username = request.data.get("username")
         password = request.data.get("password")
-        channel = get_channel_name(self.request)
         role = request.data.get("role")
-
-        _validate_channel(channel)
-        _validate_username(username)
-        _validate_password(password, self.user_model(username=username))
+        channel = get_channel_name(self.request)
 
         channel_data = {"channel_name": channel}
         if role:
-            channel_data["role"] = _validate_role(role)
+            channel_data["role"] = role
 
-        user = self.user_model.objects.create_user(username=username, password=password)
+        data = {"username": username, "password": password, "channel": channel_data}
 
-        channel_data["user"] = user
+        serializer = UserSerializer(data=data)
+        serializer.is_valid(raise_exception=True)
+        user = serializer.save()
 
-        UserChannel.objects.create(**channel_data)
-
-        user.refresh_from_db()
         data = UserSerializer(instance=user).data
 
         return ApiResponse(data=data, status=status.HTTP_201_CREATED, headers=self.get_success_headers(data))
 
     def update(self, request, *args, **kwargs):
         instance = self.get_object()
         role = request.data.get("role")
+        ui_preferences = request.data.get("ui_preferences")
+        data = {"channel": {}}
 
         if role:
-            role = _validate_role(role)
-            instance.channel.role = role
-            instance.channel.save()
+            data["channel"]["role"] = role
 
-        instance.save()
-        data = UserSerializer(instance=instance).data
+        if ui_preferences:
+            data["channel"]["ui_preferences"] = ui_preferences
 
-        return ApiResponse(data=data, status=status.HTTP_200_OK, headers=self.get_success_headers({}))
+        serializer = UserSerializer(instance=instance, data=data, partial=True)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+
+        return ApiResponse(data=serializer.data, status=status.HTTP_200_OK, headers=self.get_success_headers({}))
 
     @action(methods=["put"], detail=True, permission_classes=[IsSelf])
     def password(self, request, *args, **kwargs):
         """Allows an authenticated user to modify his own password"""
-        instance = self.get_object()
+
+        username = kwargs.get("username")
+        instance = self.user_model.objects.get(username=username)
```

Comment:
Did you test this with a username like "John Doe"? `self.get_object()` is able to decode the urlencodedentities but kwargs only contains raw values that need to be decoded manually.

Reply:
Updated the test and it seems to work yes. But still reverting to `get_object()`. For context, in certain cases there was an uncatch exception in `get_object()`. But this was fixed here https://github.com/Substra/substra-backend/pull/522/files#diff-fa54e25c9fbe9c2560d0fbef5cb51cb8f37b0c7793d322f409194417453082bbR62 so it will be catch before.